aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -0,0 +1,202 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::natives::aggregator_natives::{
+    helpers_v2::{aggregator_info_u128, aggregator_info_u64},
+    NativeAggregatorContext,
+};
+use aptos_aggregator::aggregator_extension::AggregatorID;
+use aptos_gas_schedule::gas_params::natives::aptos_framework::{
+    AGGREGATOR_V2_READ_BASE, AGGREGATOR_V2_TRY_ADD_BASE, AGGREGATOR_V2_TRY_SUB_BASE, *,
+};
+use aptos_native_interface::{
+    safely_pop_arg, RawSafeNative, SafeNativeBuilder, SafeNativeContext, SafeNativeResult,
+};
+use aptos_types::vm_status::StatusCode;
+use move_binary_format::errors::PartialVMError;
+use move_vm_runtime::native_functions::NativeFunction;
+use move_vm_types::{
+    loaded_data::runtime_types::Type,
+    values::{Struct, StructRef, Value},
+};
+use smallvec::{smallvec, SmallVec};
+use std::collections::VecDeque;
+
+/***************************************************************************************************
+ * native fun create_aggregator<Element>(limit: Element): Aggregator<Element>;
+ **************************************************************************************************/
+
+fn native_create_aggregator(
+    context: &mut SafeNativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    debug_assert_eq!(args.len(), 1);
+
+    context.charge(AGGREGATOR_V2_CREATE_AGGREGATOR_BASE)?;
+    // Get the current aggregator data.
+    let aggregator_context = context.extensions().get::<NativeAggregatorContext>();
+    let mut aggregator_data = aggregator_context.aggregator_data.borrow_mut();
+    let id = AggregatorID::ephemeral(aggregator_data.generate_id());
+
+    match ty_args[0] {
+        Type::U128 => {
+            let limit = safely_pop_arg!(args, u128);
+            aggregator_data.create_new_aggregator(id, limit);
+            Ok(smallvec![Value::struct_(Struct::pack(vec![
+                Value::u128(0),
+                Value::u128(limit),
+            ]))])
+        },
+        Type::U64 => {
+            let limit = safely_pop_arg!(args, u64);
+            aggregator_data.create_new_aggregator(id, limit as u128);
+            Ok(smallvec![Value::struct_(Struct::pack(vec![
+                Value::u64(0),
+                Value::u64(limit),
+            ]))])
+        },
+        _ => Err(PartialVMError::new(StatusCode::ABORTED)
+            .with_message("Unsupported type supplied to aggregator".to_string())
+            .with_sub_status(0x02_0005)
```

Comment:
Can we have this as a constant somewhere?

Reply:
Also, do we have a corresponding error const in the Move file?

Reply:
Done.